ssl-config/src/main/java/com/palantir/remoting/ssl/pkcs1/Pkcs1Readers.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting.ssl.pkcs1;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.BaseEncoding;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.spec.RSAPrivateKeySpec;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class Pkcs1Readers {
+
+    private static final List<Pkcs1Reader> PKCS1_READERS = ImmutableList.of(
+            new BouncyCastlePkcs1Reader(),
+            new SunPkcs1Reader());
+
+    private Pkcs1Readers() {
+    }
+
+    public static RSAPrivateKeySpec readPrivateKey(String privateKeyString)
+            throws GeneralSecurityException, IOException {
+        byte[] privateKeyDerBytes = BaseEncoding.base64().decode(privateKeyString);
+
+        // try reading private key using all known readers
+        List<String> missingPackages = new ArrayList<>();
+        for (Pkcs1Reader reader : PKCS1_READERS) {
+            try {
+                // return private key if it is read successfully
+                return reader.readPrivateKey(privateKeyDerBytes);
+            } catch (GeneralSecurityException e) {
```

Comment:
I was thinking:
 * Pcks1Reader interface defined in ssl-config
 * BouncyCastlePcks1Reader class defined in pcks1-reader-bouncy-castle (depends explicitly on bouncy castle, also depends on ssl-config)
 * SunPcks1Reader class defined in pckds1-reader-sun (depends on ssl-config)

Which limits this to just two extra projects instead of 3. Pcks1Reader is really only relevant to ssl-config anyway, for now, at least.

Thoughts?